E0277 - the trait bound `(core::ops::Bound<_>, core::ops::Bound<()>): core::ops::RangeBounds<wire::ip::Cidr>` is not satisfied

Hi,

Looks like 0.6 does not compile on rustc 1.45 :

```
error[E0277]: the trait bound `(core::ops::Bound<_>, core::ops::Bound<()>): core::ops::RangeBounds<wire::ip::Cidr>` is not satisfied
   --> /home/user/.cargo/registry/src/github.com-1ecc6299db9ec823/smoltcp-0.6.0/src/iface/route.rs:120:45
    |
120 |         for (prefix, route) in self.storage.range((Bound::Unbounded, Bound::Included(cidr))).rev() {
    |                                             ^^^^^ the trait `core::ops::RangeBounds<wire::ip::Cidr>` is not implemented for `(core::ops::Bound<_>, core::ops::Bound<()>)`
    |
    = help: the following implementations were found:
              <(core::ops::Bound<&'a T>, core::ops::Bound<&'a T>) as core::ops::RangeBounds<T>>
              <(core::ops::Bound<T>, core::ops::Bound<T>) as core::ops::RangeBounds<T>>

error: aborting due to previous error

For more information about this error, try `rustc --explain E0277`.
error: could not compile `smoltcp`.
```

Looks like it's doing the same thing on 1.43, am I doing something wrong or is there a supported version of rust I should be using ?

Thank you

--- a/src/iface/route.rs
+++ b/src/iface/route.rs
@@ -117,7 +117,7 @@ impl<'a> Routes<'a> {
             _ => unimplemented!()
         };
 
-        for (prefix, route) in self.storage.range((Bound::Unbounded, Bound::Included(cidr))).rev() {
+        for (prefix, route) in self.storage.range((Bound::Unbounded::<IpCidr>, Bound::Included(cidr))).rev() {
             // TODO: do something with route.preferred_until
             if let Some(expires_at) = route.expires_at {
                 if timestamp > expires_at {

